pkg/test/framework/components/galley/snapshot.go

```diff
@@ -48,13 +48,31 @@ type SnapshotValidatorFunc func(actuals []*SnapshotObject) error
 // is found in the snapshot.
 func NewSingleObjectSnapshotValidator(ns string, fn func(ns string, actual *SnapshotObject) error) SnapshotValidatorFunc {
 	return func(actuals []*SnapshotObject) error {
-		if len(actuals) != 1 {
+		filteredActuals := actualsByNamespace(ns, actuals)
+		if len(filteredActuals) != 1 {
 			return fmt.Errorf("expected 1 resource, found %d", len(actuals))
 		}
 		return fn(ns, actuals[0])
 	}
 }
 
+func actualsByNamespace(ns string, actuals []*SnapshotObject) (result []*SnapshotObject) {
```

Comment:
nit: can we drop the `actuals` from the method name? Maybe `getForNamespace`, or something similar?